assets/src/stories-editor/components/rotatable-box/index.js

```diff
@@ -43,45 +48,45 @@ class RotatableBox extends Component {
 		const { keyCode } = e;
 
 		if ( ESCAPE === keyCode ) {
-			this.elementRef.classList.remove( 'is-rotating' );
-			this.elementRef.style.transform = `rotate(${ this.props.initialAngle }deg)`;
-
-			this.setState(
-				{
-					isRotating: false,
-					angle: this.props.initialAngle,
-				},
-				() => this.props.onRotateStop && this.props.onRotateStop( e, this.props.initialAngle )
-			);
+			elementRef.classList.remove( 'is-rotating' );
+			elementRef.style.transform = `rotate(${ initialAngle }deg)`;
+
+			setIsRotating( false );
+			setAngle( initialAngle );
+
+			if ( onRotateStop ) {
+				onRotateStop( e, initialAngle );
+			}
 		} else if ( LEFT === keyCode || RIGHT === keyCode ) {
-			let angle = LEFT === keyCode ? this.state.angle - 30 : this.state.angle + 30;
-			if ( angle > 360 ) {
-				angle -= 360;
-			} else if ( angle <= -360 ) {
-				angle += 360;
+			let newAngle = LEFT === keyCode ? currentAngle - 30 : currentAngle + 30;
+			if ( newAngle > 360 ) {
+				newAngle -= 360;
+			} else if ( newAngle <= -360 ) {
+				newAngle += 360;
 			}
 
-			this.elementRef.style.transform = `rotate(${ angle }deg)`;
+			elementRef.style.transform = `rotate(${ newAngle }deg)`;
```

Comment:
I'd do this in a `useEffect` with `angle` as a guard. Then you don't need it both here and in the `onMouseMove`.